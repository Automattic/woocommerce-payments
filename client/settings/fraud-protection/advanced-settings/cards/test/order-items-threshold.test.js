/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
/**
 * Internal dependencies
 */
import FraudPreventionSettingsContext from '../../context';
import OrderItemsThresholdRuleCard, {
	OrderItemsThresholdValidation,
} from '../order-items-threshold';

describe( 'Order items threshold card', () => {
	const settings = {
		order_items_threshold: {
			enabled: false,
			block: false,
			min_items: null,
			max_items: null,
		},
	};
	const setSettings = jest.fn();
	const contextValue = {
		advancedFraudProtectionSettings: settings,
		setAdvancedFraudProtectionSettings: setSettings,
	};
	test( 'renders correctly', () => {
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders correctly when enabled', () => {
		settings.order_items_threshold.enabled = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders correctly when enabled and checked', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders like disabled when checked, but not enabled', () => {
		settings.order_items_threshold.enabled = false;
		settings.order_items_threshold.block = true;
		const { container } = render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect( container ).toMatchSnapshot();
	} );
	test( 'renders warning when both fields are empty', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect(
			screen.queryByText(
				'An item range must be set for this filter to take effect.'
			)
		).toBeInTheDocument();
	} );
	test( "doesn't render warning when only min items field is filled", () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		settings.order_items_threshold.min_items = 1;
		settings.order_items_threshold.max_items = null;
		render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect(
			screen.queryByText(
				'An item range must be set for this filter to take effect.'
			)
		).not.toBeInTheDocument();
	} );
	test( "doesn't render warning when only max items field is filled", () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		settings.order_items_threshold.min_items = null;
		settings.order_items_threshold.max_items = 1;
		render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect(
			screen.queryByText(
				'An item range must be set for this filter to take effect.'
			)
		).not.toBeInTheDocument();
	} );
	test( "doesn't render warning when both fields are filled", () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		settings.order_items_threshold.min_items = 1;
		settings.order_items_threshold.max_items = 2;
		render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect(
			screen.queryByText(
				'An item range must be set for this filter to take effect.'
			)
		).not.toBeInTheDocument();
	} );
	test( 'renders error when min items is greater than max items', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = true;
		settings.order_items_threshold.min_items = 2;
		settings.order_items_threshold.max_items = 1;
		render(
			<FraudPreventionSettingsContext.Provider value={ contextValue }>
				<OrderItemsThresholdRuleCard />
			</FraudPreventionSettingsContext.Provider>
		);
		expect(
			screen.queryByText(
				'An item range must be set for this filter to take effect.'
			)
		).not.toBeInTheDocument();
		expect(
			screen.queryByText(
				'Maximum item count must be greater than the minimum item count.'
			)
		).toBeInTheDocument();
	} );

	test( 'validation returns true when `min_items` is set', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = null;
		settings.order_items_threshold.min_items = 100;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( true );
		expect( setValidationError.mock.calls.length ).toBe( 0 );
	} );
	test( 'validation returns true when `max_amount` is set', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = 100;
		settings.order_items_threshold.min_items = null;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( true );
		expect( setValidationError.mock.calls.length ).toBe( 0 );
	} );
	test( 'validation returns true when both are set', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = 100;
		settings.order_items_threshold.min_items = 10;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( true );
		expect( setValidationError.mock.calls.length ).toBe( 0 );
	} );
	test( 'validation returns true when setting is not enabled', () => {
		settings.order_items_threshold.enabled = false;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = null;
		settings.order_items_threshold.min_items = null;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( true );
		expect( setValidationError.mock.calls.length ).toBe( 0 );
	} );
	test( 'validation returns false when amounts are not set', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = null;
		settings.order_items_threshold.min_items = null;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( false );
		expect( setValidationError.mock.calls.length ).toBe( 1 );
		expect( setValidationError.mock.calls[ 0 ] ).toContain(
			'An item range must be set for the "Order Item Threshold" filter.'
		);
	} );
	test( 'validation returns false when min amount is greater than max amount', () => {
		settings.order_items_threshold.enabled = true;
		settings.order_items_threshold.block = false;
		settings.order_items_threshold.max_items = 10;
		settings.order_items_threshold.min_items = 100;
		const setValidationError = jest.fn();
		const validationResult = OrderItemsThresholdValidation(
			settings.order_items_threshold,
			setValidationError
		);
		expect( validationResult ).toBe( false );
		expect( setValidationError.mock.calls.length ).toBe( 1 );
		expect( setValidationError.mock.calls[ 0 ] ).toContain(
			'Maximum item count must be greater than the minimum item count on the "Order Item Threshold" rule.'
		);
	} );
} );
