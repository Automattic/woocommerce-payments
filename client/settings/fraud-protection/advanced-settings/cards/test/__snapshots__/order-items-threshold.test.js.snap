// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order items threshold card renders correctly 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          Order Items Threshold
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter compares the amount of items in an order to the minimum and maximum counts that you specify.
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <div
          class="fraud-protection-rule-toggle"
        >
          <strong>
            Enable filtering
          </strong>
          <div
            class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
          >
            <div
              class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
            >
              <span
                class="components-form-toggle"
              >
                <input
                  aria-describedby="inspector-toggle-control-0__help"
                  class="components-form-toggle__input"
                  id="inspector-toggle-control-0"
                  type="checkbox"
                />
                <span
                  class="components-form-toggle__track"
                />
                <span
                  class="components-form-toggle__thumb"
                />
              </span>
              <label
                class="components-toggle-control__label"
                for="inspector-toggle-control-0"
              >
                Screen transactions for abnormal item counts
              </label>
            </div>
            <p
              class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
              id="inspector-toggle-control-0__help"
            >
              When enabled, the payment method will not be charged until you review and approve the transaction
            </p>
          </div>
        </div>
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            An unusually high item count, compared to the average for your business, can indicate potential fraudulent activity.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Order items threshold card renders correctly when enabled 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          Order Items Threshold
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter compares the amount of items in an order to the minimum and maximum counts that you specify.
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <div
          class="fraud-protection-rule-toggle"
        >
          <strong>
            Enable filtering
          </strong>
          <div
            class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
          >
            <div
              class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
            >
              <span
                class="components-form-toggle is-checked"
              >
                <input
                  aria-describedby="inspector-toggle-control-1__help"
                  class="components-form-toggle__input"
                  id="inspector-toggle-control-1"
                  type="checkbox"
                />
                <span
                  class="components-form-toggle__track"
                />
                <span
                  class="components-form-toggle__thumb"
                />
              </span>
              <label
                class="components-toggle-control__label"
                for="inspector-toggle-control-1"
              >
                Screen transactions for abnormal item counts
              </label>
            </div>
            <p
              class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
              id="inspector-toggle-control-1__help"
            >
              When enabled, the payment method will not be charged until you review and approve the transaction
            </p>
          </div>
          <div>
            <div
              class="fraud-protection-rule-toggle-children-container"
            >
              <strong>
                Limits
              </strong>
              <div
                class="fraud-protection-rule-toggle-children-horizontal-form"
              >
                <div
                  class="fraud-protection-rule-toggle-children-vertical-form"
                >
                  <label
                    for="fraud-protection-purchase-price-minimum"
                  >
                    Minimum items per order
                  </label>
                  <div
                    class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
                  >
                    <div
                      class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                    >
                      <input
                        aria-describedby="inspector-text-control-0__help"
                        class="components-text-control__input"
                        id="fraud-protection-purchase-price-minimum"
                        placeholder="0"
                        type="text"
                        value="0"
                      />
                    </div>
                    <p
                      class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
                      id="inspector-text-control-0__help"
                    >
                      Leave blank for no limit
                    </p>
                  </div>
                </div>
                <div
                  class="fraud-protection-rule-toggle-children-vertical-form"
                >
                  <label
                    for="fraud-protection-purchase-price-maximum"
                  >
                    Maximum items per order
                  </label>
                  <div
                    class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
                  >
                    <div
                      class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                    >
                      <input
                        aria-describedby="inspector-text-control-1__help"
                        class="components-text-control__input"
                        id="fraud-protection-purchase-price-maximum"
                        placeholder="0"
                        prefix="$"
                        type="text"
                        value="0"
                      />
                    </div>
                    <p
                      class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
                      id="inspector-text-control-1__help"
                    >
                      Leave blank for no limit
                    </p>
                  </div>
                </div>
              </div>
              <div>
                <br />
                <div
                  class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning"
                >
                  <div
                    class="fraud-protection-rule-card-notice-icon"
                  >
                    <svg
                      data-testid="rule-card-notice-warning-icon-svg"
                      fill="none"
                      height="18"
                      viewBox="0 0 18 18"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M9 1.75C4.99594 1.75 1.75 4.99594 1.75 9C1.75 13.0041 4.99594 16.25 9 16.25C13.0041 16.25 16.25 13.0041 16.25 9C16.25 4.99594 13.0041 1.75 9 1.75ZM0.25 9C0.25 4.16751 4.16751 0.25 9 0.25C13.8325 0.25 17.75 4.16751 17.75 9C17.75 13.8325 13.8325 17.75 9 17.75C4.16751 17.75 0.25 13.8325 0.25 9Z"
                        fill="#bd8600"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M10 4H8V10H10V4Z"
                        fill="#bd8600"
                      />
                      <path
                        d="M10 12H8V14H10V12Z"
                        fill="#bd8600"
                      />
                    </svg>
                  </div>
                  <div
                    class="fraud-protection-rule-card-notice-text"
                  >
                    An item range must be set for this filter to take effect.
                  </div>
                </div>
              </div>
            </div>
            <strong>
              Advanced
            </strong>
            <div
              class="fraud-protection-rule-toggle-checkbox-container"
            >
              <div
                class="components-base-control fraud-protection-rule-toggle-checkbox css-wdf2ti-Wrapper e1puf3u0"
              >
                <div
                  class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                >
                  <span
                    class="components-checkbox-control__input-container"
                  >
                    <input
                      class="components-checkbox-control__input"
                      id="inspector-checkbox-control-0"
                      type="checkbox"
                      value="1"
                    />
                  </span>
                  <label
                    class="components-checkbox-control__label"
                    for="inspector-checkbox-control-0"
                  >
                    Block Payment
                  </label>
                </div>
              </div>
              <button
                class="wcpay-tooltip__content-wrapper"
                type="button"
              >
                <div
                  class="wcpay-tooltip__content-wrapper"
                >
                  <div
                    class="fraud-protection-rule-toggle-checkbox-container-help"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="16"
                      role="img"
                      viewBox="3 3 18 18"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.75 10.25C9.75 9.00736 10.7574 8 12 8C13.2426 8 14.25 9.00736 14.25 10.25C14.25 														11.4083 13.3748 12.3621 12.2496 12.4863C12.1124 12.5015 12 12.6119 12 12.75V14M12 														15V16.5M20 12C20 16.4183 16.4183 20 12 20C7.58172 20 4 16.4183 4 12C4 7.58172 														7.58172 4 12 4C16.4183 4 20 7.58172 20 12Z"
                        stroke="#1E1E1E"
                        stroke-width="1.5"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            An unusually high item count, compared to the average for your business, can indicate potential fraudulent activity.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Order items threshold card renders correctly when enabled and checked 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          Order Items Threshold
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter compares the amount of items in an order to the minimum and maximum counts that you specify.
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <div
          class="fraud-protection-rule-toggle"
        >
          <strong>
            Enable filtering
          </strong>
          <div
            class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
          >
            <div
              class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
            >
              <span
                class="components-form-toggle is-checked"
              >
                <input
                  aria-describedby="inspector-toggle-control-2__help"
                  class="components-form-toggle__input"
                  id="inspector-toggle-control-2"
                  type="checkbox"
                />
                <span
                  class="components-form-toggle__track"
                />
                <span
                  class="components-form-toggle__thumb"
                />
              </span>
              <label
                class="components-toggle-control__label"
                for="inspector-toggle-control-2"
              >
                Screen transactions for abnormal item counts
              </label>
            </div>
            <p
              class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
              id="inspector-toggle-control-2__help"
            >
              When enabled, the payment method will not be charged until you review and approve the transaction
            </p>
          </div>
          <div>
            <div
              class="fraud-protection-rule-toggle-children-container"
            >
              <strong>
                Limits
              </strong>
              <div
                class="fraud-protection-rule-toggle-children-horizontal-form"
              >
                <div
                  class="fraud-protection-rule-toggle-children-vertical-form"
                >
                  <label
                    for="fraud-protection-purchase-price-minimum"
                  >
                    Minimum items per order
                  </label>
                  <div
                    class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
                  >
                    <div
                      class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                    >
                      <input
                        aria-describedby="inspector-text-control-2__help"
                        class="components-text-control__input"
                        id="fraud-protection-purchase-price-minimum"
                        placeholder="0"
                        type="text"
                        value="0"
                      />
                    </div>
                    <p
                      class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
                      id="inspector-text-control-2__help"
                    >
                      Leave blank for no limit
                    </p>
                  </div>
                </div>
                <div
                  class="fraud-protection-rule-toggle-children-vertical-form"
                >
                  <label
                    for="fraud-protection-purchase-price-maximum"
                  >
                    Maximum items per order
                  </label>
                  <div
                    class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
                  >
                    <div
                      class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                    >
                      <input
                        aria-describedby="inspector-text-control-3__help"
                        class="components-text-control__input"
                        id="fraud-protection-purchase-price-maximum"
                        placeholder="0"
                        prefix="$"
                        type="text"
                        value="0"
                      />
                    </div>
                    <p
                      class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
                      id="inspector-text-control-3__help"
                    >
                      Leave blank for no limit
                    </p>
                  </div>
                </div>
              </div>
              <div>
                <br />
                <div
                  class="fraud-protection-rule-card-notice fraud-protection-rule-card-notice-warning"
                >
                  <div
                    class="fraud-protection-rule-card-notice-icon"
                  >
                    <svg
                      data-testid="rule-card-notice-warning-icon-svg"
                      fill="none"
                      height="18"
                      viewBox="0 0 18 18"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M9 1.75C4.99594 1.75 1.75 4.99594 1.75 9C1.75 13.0041 4.99594 16.25 9 16.25C13.0041 16.25 16.25 13.0041 16.25 9C16.25 4.99594 13.0041 1.75 9 1.75ZM0.25 9C0.25 4.16751 4.16751 0.25 9 0.25C13.8325 0.25 17.75 4.16751 17.75 9C17.75 13.8325 13.8325 17.75 9 17.75C4.16751 17.75 0.25 13.8325 0.25 9Z"
                        fill="#bd8600"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M10 4H8V10H10V4Z"
                        fill="#bd8600"
                      />
                      <path
                        d="M10 12H8V14H10V12Z"
                        fill="#bd8600"
                      />
                    </svg>
                  </div>
                  <div
                    class="fraud-protection-rule-card-notice-text"
                  >
                    An item range must be set for this filter to take effect.
                  </div>
                </div>
              </div>
            </div>
            <strong>
              Advanced
            </strong>
            <div
              class="fraud-protection-rule-toggle-checkbox-container"
            >
              <div
                class="components-base-control fraud-protection-rule-toggle-checkbox css-wdf2ti-Wrapper e1puf3u0"
              >
                <div
                  class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
                >
                  <span
                    class="components-checkbox-control__input-container"
                  >
                    <input
                      checked=""
                      class="components-checkbox-control__input"
                      id="inspector-checkbox-control-1"
                      type="checkbox"
                      value="1"
                    />
                    <svg
                      aria-hidden="true"
                      class="components-checkbox-control__checked"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M18.3 5.6L9.9 16.9l-4.6-3.4-.9 1.2 5.8 4.3 9.3-12.6z"
                      />
                    </svg>
                  </span>
                  <label
                    class="components-checkbox-control__label"
                    for="inspector-checkbox-control-1"
                  >
                    Block Payment
                  </label>
                </div>
              </div>
              <button
                class="wcpay-tooltip__content-wrapper"
                type="button"
              >
                <div
                  class="wcpay-tooltip__content-wrapper"
                >
                  <div
                    class="fraud-protection-rule-toggle-checkbox-container-help"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="16"
                      role="img"
                      viewBox="3 3 18 18"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.75 10.25C9.75 9.00736 10.7574 8 12 8C13.2426 8 14.25 9.00736 14.25 10.25C14.25 														11.4083 13.3748 12.3621 12.2496 12.4863C12.1124 12.5015 12 12.6119 12 12.75V14M12 														15V16.5M20 12C20 16.4183 16.4183 20 12 20C7.58172 20 4 16.4183 4 12C4 7.58172 														7.58172 4 12 4C16.4183 4 20 7.58172 20 12Z"
                        stroke="#1E1E1E"
                        stroke-width="1.5"
                      />
                    </svg>
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            An unusually high item count, compared to the average for your business, can indicate potential fraudulent activity.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Order items threshold card renders like disabled when checked, but not enabled 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection-rule-card css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body fraud-protection-rule-card-header-container css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <p
          class="fraud-protection-rule-card-header"
        >
          Order Items Threshold
        </p>
        <p
          class="fraud-protection-rule-card-description"
        >
          This filter compares the amount of items in an order to the minimum and maximum counts that you specify.
        </p>
      </div>
    </div>
    <hr />
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div>
        <div
          class="fraud-protection-rule-toggle"
        >
          <strong>
            Enable filtering
          </strong>
          <div
            class="components-base-control components-toggle-control fraud-protection-rule-toggle-toggle css-wdf2ti-Wrapper e1puf3u0"
          >
            <div
              class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
            >
              <span
                class="components-form-toggle"
              >
                <input
                  aria-describedby="inspector-toggle-control-3__help"
                  class="components-form-toggle__input"
                  id="inspector-toggle-control-3"
                  type="checkbox"
                />
                <span
                  class="components-form-toggle__track"
                />
                <span
                  class="components-form-toggle__thumb"
                />
              </span>
              <label
                class="components-toggle-control__label"
                for="inspector-toggle-control-3"
              >
                Screen transactions for abnormal item counts
              </label>
            </div>
            <p
              class="components-base-control__help css-1wm1a55-StyledHelp e1puf3u3"
              id="inspector-toggle-control-3__help"
            >
              When enabled, the payment method will not be charged until you review and approve the transaction
            </p>
          </div>
        </div>
        <div
          class="fraud-protection-rule-description"
        >
          <strong>
            How does this filter protect me?
          </strong>
          <p>
            An unusually high item count, compared to the average for your business, can indicate potential fraudulent activity.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
